Automattic/wp-calypso
# Stepper: Prevent track `calypso_signup_start` multiple times

Closes #93950
Related to [#91280](https://github.com/Automattic/wp-calypso/issues/91280)

## Proposed Changes

* ~Add useDisposableEffect hook. It enables the trigger or not of a use effect function once and programmatically.~
* ~Update useSignUpStartTracking to use this new hook~
* Ignore the changes incoming from the useSignupStartEventProps after the event is triggered



## Why are these changes being made?
*  We have the extra track event calls on the stepper because the useSignupStartEventProps always returns a new object instance.  It is causing new calls on the [this](https://github.com/Automattic/wp-calypso/pull/94248/files#diff-1c00d7502e68a6cb680dfbf3bf9d6249da1a22f8bcf51ade0ce1574067597e64R42) `useEffect`  and as consequence new `calypso_signup_start` calls. 


## Testing Instructions
Scenario 1: Regular tracking
* Go to any signup flow which uses  the `useSignupStartEventProps` 
* Use Vigilant watch and check if only one calypso_signup_start was triggered.
* Check no  `calypso_signup_start` was triggered. 


Scenario 2: `start` query param
* Go to any signup flow non-initial step using the signup query param  (E.g, `/setup/migration/create-site?start=true`) 
* Use Vigilant watch and check if only one calypso_signup_start was triggered.
* Follow the steps until it arrives on the checkout page
* Check no  calypso_signup_start was triggered. 


## Pre-merge Checklist
